src/index.js

```diff
@@ -73,59 +84,73 @@ class Masq {
     this.profile = id
   }
 
+  _getDB () {
+    if (!this.profile) throw Error('No profile selected')
+    let db = this.dbs[this.profile]
+    if (!db) throw Error('db does not exist for selected profile')
+    return db
+  }
+
   /**
    * Get a value
    * @param {string} key - Key
+   * @returns {Promise}
    */
   get (key) {
-    return new Promise((resolve, reject) => {
-      if (!this.profile) return reject(Error('No profile selected'))
-      const db = this.dbs[this.profile]
-      db.get(key, (err, nodes) => {
-        if (err) return reject(err)
-        if (!nodes.length) return resolve(nodes[0])
-        resolve(nodes[0].value)
+    try {
+      let db = this._getDB()
```

Comment:
you should not need to wrap this with try/catch, if you move `_getDB()` inside the promise and it throws, then `get()` will reject the error too (

Reply:
https://mrcoles.com/javascript-promises-and-errors/

Reply:
fixed